src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,40 @@
 package mate.academy;
 
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private static final int TIMEOUT = 10;
+    private final CopyOnWriteArrayList<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executorService = Executors.newFixedThreadPool(10);
 
+    public void registerListener(EventListener listener) {
+        listeners.addIfAbsent(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        executorService.submit(() -> {
+            for (EventListener listener : listeners) {
+                listener.onEvent(event);
+            }
+        });
```

Comment:
When submitting tasks to an ExecutorService, it's important to handle possible rejections, especially during shutdown. Consider adding a RejectedExecutionHandler or handling rejected tasks to avoid losing any event notifications.

Reply:
Not resolved